docs/source/serving/service-yaml-spec.rst

```diff
@@ -45,14 +45,11 @@ Available fields:
         # threshold specified below.
         max_replicas: 3
         # Following thresholds describe when to scale up or down.
-        # QPS threshold for scaling up (optional). If the QPS of your service
-        # exceeds this threshold, SkyServe will scale up your service by one
-        # replica. If not specified, SkyServe will **NOT** scale up your service.
-        qps_upper_threshold: 10
-        # QPS threshold for scaling down (optional). If the QPS of your service
-        # is below this threshold, SkyServe will scale down your service by one
-        # replica. If not specified, SkyServe will **NOT** scale down your service.
-        qps_lower_threshold: 2
+        # Target QPS per replica (optional). If the QPS of your service
```

Comment:
I feel like we should expose all the config to user for fine-grained control - probably sth like this:

```yaml
# Following specs describe the autoscaling policy.
# Target request per second per replica (optional). SkyServe will scale your
# service so that, ultimately, each replica manages approximately
# target_rps_per_replica queries per second. **Autoscaling will only be
# enabled if this value is specified.**
target_rps_per_replica: 5
# Autoscaling decision interval in seconds (optional). Defaults to 20 seconds.
# Every autoscaling_decision_interval seconds, SkyServe will check the QPS
# of your service and make autoscaling decisions. Target number of replicas
# will be calculated by:
#   target_replicas = ceil(current_qps / target_rps_per_replica)
# This value will also be capped by min_replicas and max_replicas.
autoscaling_decision_interval: 20
# Upscale and downscale delay in seconds (optional). Defaults to 300 seconds
# (5 minutes) and 1200 seconds (20 minutes) respectively. To avoid aggressive
# autoscaling, SkyServe will only upscale or downscale your service if the
# QPS of your service is higher or lower than the target QPS for a period
# of time. This period of time is controlled by upscale_delay_seconds and
# downscale_delay_seconds. If you want to scale your service more aggressively,
# you can set these values to a smaller number.
upscale_delay_seconds: 300
downscale_delay_seconds: 1200
```

Reply:
sounds good!